src/qgraphicsmozview_p.cpp

```diff
@@ -425,6 +431,33 @@ void QGraphicsMozViewPrivate::UpdateViewSize()
         return;
     }
 
+    if (mOrientationDirty) {
+        QMatrix affine;
+        ScreenRotation rotation = ROTATION_0;
+        switch (mOrientation) {
+        case Qt::LandscapeOrientation:
+            rotation = mozilla::ROTATION_90;
+            affine.rotate(90);
+            affine.translate(0, -mGLSurfaceSize.height());
+            break;
+        case Qt::InvertedLandscapeOrientation:
+            rotation = mozilla::ROTATION_270;
+            affine = affine.rotate(270);
+            affine.translate(-mGLSurfaceSize.width(), 0);
+            break;
+        case Qt::InvertedPortraitOrientation:
+            rotation = mozilla::ROTATION_180;
+            affine = affine.rotate(180);
+            affine.translate(-mGLSurfaceSize.width(), -mGLSurfaceSize.height());
+            break;
+        default:
+            break;
+        }
+        gfxMatrix matr(affine.m11(), affine.m12(), affine.m21(), affine.m22(), affine.dx(), affine.dy());
+        mView->SetScreenRotation(rotation, matr);
```

Comment:
My point is that affine matrix calculations could be done behind SetScreenRotation API, because the only input for affine calculation we use is: rotation value and mGLSurfaceSize